app/mixins/route/account/billing.js

```diff
@@ -20,4 +28,22 @@ export default Mixin.create({
       creditCardInfo,
     });
   },
+
+  setupBillingStepSubscriptions(controllerName) {
+    const controller = this.controllerFor(controllerName);
+    controller.addObserver('billingStep', this, 'handleBillingStepChange');
```

Comment:
Observers are considered a bad practice in Ember, as they contradict DDAU approach and quite unpredictable and hard to debug. But I think, changing query parameter update to navigation should make this observation unnecessary.